src/notification/mod.rs

```diff
@@ -417,6 +349,135 @@ pub async fn register(
     }
 }
 
+pub async fn send(
+    State(state): State<AppState>,
+    Json(req): Json<NotificationSendReq>,
+) -> impl IntoResponse {
+    let payload = req.payload;
+    let signature = req.signature;
+    if let Err(e) = util::validate_npub(&payload.receiver) {
+        error!("notification send with invalid receiver npub: {e}");
+        return (
+            StatusCode::BAD_REQUEST,
+            Json(ErrorResp::new("Invalid receiver npub")),
+        )
+            .into_response();
+    }
+
+    let x_only_sender = match util::validate_npub(&payload.sender) {
+        Ok(n) => n,
+        Err(e) => {
+            error!("notification send with invalid sender npub: {e}");
+            return (
+                StatusCode::BAD_REQUEST,
+                Json(ErrorResp::new("Invalid sender npub")),
+            )
+                .into_response();
+        }
+    };
+
+    let notification_type = match PreferencesFlags::from_name(&payload.kind) {
+        Some(nt) => nt,
+        None => {
+            error!(
+                "notification send with invalid event type: {}",
+                &payload.kind
+            );
+            return (
+                StatusCode::BAD_REQUEST,
+                Json(ErrorResp::new("Invalid kind")),
+            )
+                .into_response();
+        }
+    };
+
+    if payload.id.is_empty() || !payload.id.starts_with(BITCR_PREFIX) {
+        error!("notification send with empty, or invalid id",);
+        return (StatusCode::BAD_REQUEST, Json(ErrorResp::new("Invalid ID"))).into_response();
+    }
+
+    // make sure sender signed the requesed
+    match util::verify_request(&payload, &signature, &x_only_sender) {
+        Ok(true) => {
+            let email_preferences = match state
+                .notification_store
+                .get_email_preferences_for_npub(&payload.receiver)
```

Comment:
Reading through the PR I realized that we probably want to have a whitelist of npubs that are allowed to send us notifications :thinking:

Reply:
Good point!
I think there's multiple ways we can go about it (similar to the notifications in E-Bill directly).

A problem with the whitelist would be keeping it up2date vs. the danger of missing notifications.

I think we'll have to play around with the initial implementation and see what makes sense, also in terms of spam protection etc.